internal/pkg/bootstrap/handlers/database/database.go

```diff
@@ -100,11 +101,22 @@ func (d Database) BootstrapHandler(
 
 	loggingClient := container.LoggingClientFrom(dic.Get)
 
+	// get database credentials.
+	var credentials config.Credentials
+	for startupTimer.HasNotElapsed() {
+		var err error
+		credentials, err = container.CredentialsProviderFrom(dic.Get).GetDatabaseCredentials(d.database.GetDatabaseInfo()["Primary"])
+		if err == nil {
+			break
+		}
+		startupTimer.SleepForInterval()
```

Comment:
You should log the error when err != nil -- similar to what's being done below when d.NewDBClient() returns an err.